Editor/ProjectIssue.cs

```diff
@@ -43,33 +47,53 @@ public static IssueBuilder Create(IssueCategory category, string description)
         [SerializeField] string[] m_CustomProperties;
         [SerializeField] Severity m_Severity;
 
-        internal ProjectIssue(IssueCategory category, Descriptor descriptor, params object[] args)
+        /// <summary>
+        /// Constructs and returns an instance of ProjectIssue
+        /// </summary>
+        /// <param name="category">Issue category</param>
+        /// <param name="descriptor">Diagnostic descriptor</param>
+        /// <param name="args">Arguments to be used in the message formatting</param>
+        public ProjectIssue(IssueCategory category, Descriptor descriptor, params object[] args)
         {
             m_Descriptor = descriptor;
             m_Description = args.Length > 0 ? string.Format(descriptor.messageFormat, args) : descriptor.title;
             m_Category = category;
             m_Severity = descriptor.defaultSeverity;
         }
 
-        internal ProjectIssue(IssueCategory category, string description)
+        /// <summary>
+        /// Constructs and returns an instance of ProjectIssue
+        /// </summary>
+        /// <param name="category">Issue category</param>
+        /// <param name="description">Issue description</param>
+        public ProjectIssue(IssueCategory category, string description)
         {
             m_Description = description;
             m_Category = category;
             m_Severity = Severity.Default;
         }
 
+        /// <summary>
+        /// This issue's category
+        /// </summary>
         public IssueCategory category => m_Category;
 
+        /// <summary>
+        /// Custom properties
+        /// </summary>
         public string[] customProperties
         {
             get => m_CustomProperties;
-            internal set => m_CustomProperties = value;
+            /*public*/ set => m_CustomProperties = value;
```

Comment:
Just double-checking:  Here and in a few lines below "/*public*/" was added to the setter, not the getter.

Shall we leave them anyway to point out that this is implicitly public or rather remove all occurances?